cli/src/commands/git/clone.rs

```diff
@@ -196,12 +230,13 @@ fn configure_remote(
     workspace_command: WorkspaceCommandHelper,
     remote_name: &RemoteName,
     source: &str,
+    fetch_tags: gix::remote::fetch::Tags,
 ) -> Result<WorkspaceCommandHelper, CommandError> {
     git::add_remote(
         workspace_command.repo().store(),
         remote_name,
         source,
-        Default::default(),
+        fetch_tags,
```

Comment:
Related to this, I think the default runtime behavior of `jj git clone` should be `Tags::All`. The configuration should be `Tags::Included`, though.

https://github.com/jj-vcs/jj/issues/4682

(This is a separate issue, so can be addressed later.)

Reply:
@yuja rereading your comment and I want to make sure I understand your feedback:

> Related to this, I think the default runtime behavior of jj git clone should be Tags::All.

This part makes sense to change the CLI args to default to `Tags::All`

> The configuration should be Tags::Included, though.

This part I don't follow: are you suggesting we keep the CLI defaults to `Tags::Included` _for this PR_ and then change the default to `Tags::All` in a later PR? Or do you mean change `Default::default()` to `Tags::Included` to be explicit for now? Or something else?

Reply:
I mean:

* `jj git clone` should temporarily do fetch refs with `Tags::All` because that's what `git clone` would do. It's implementation detail that we do `git init && git fetch` internally.
* The default should stay `Tags::Included` because that's the default in Git.

Reply:
Ah thank you for clarifying! I had misunderstood the fact that `git clone` will always clone tags (and that there's no way of only cloning "included" tags)

I've reworked #7192 to correctly fetch all tags during cloning (but otherwise allow configuring the remote's behavior for subsequent fetches). We can move the rest of the conversation on that implementation there but for now I think this is ready for a re-review!